gordo_components/client/client.py

```diff
@@ -128,6 +129,16 @@ def __init__(
             ignore_unhealthy_targets=ignore_unhealthy_targets,
         )
 
+    @property
+    def endpoints(self) -> typing.List[EndpointMetadata]:
+        with self._mutex:
+            return self._endpoints
+
+    @endpoints.setter
+    def endpoints(self, endpoints: typing.List[EndpointMetadata]):
+        with self._mutex:
```

Comment:
Are you sure this works as intended? I would guess what would happen with 100 threads is that there would be 100 requests to watchman, and then each one of them gets written sequentially to the endpoints variable. But becuase of GIL python, the latter would happen anyway :-p Where the lock is needed is to ensure that only a single instance calls watchman.

Reply:
So I would think the lock is only needed around line 125, together with a `if self.endpoint` check.

Reply:
Spot on, my dear Watson

Reply:
I always saw myself more as the Sherlock...